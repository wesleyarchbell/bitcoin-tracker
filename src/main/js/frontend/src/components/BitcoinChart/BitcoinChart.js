import React, { Component } from 'react';
import { Line } from 'react-chartjs-2'; 

class BitcoinChart extends Component {

    constructor(props) {
        super(props);
        this.state = {
            chartData: {}
        }
    }

    componentDidUpdate() {        
        if (this.chartReference) {
            this.chartReference.props.data.labels = this.props.data.map(d => d.time);
            this.chartReference.props.data.datasets[0].data = this.props.data.map(d => d.price);
        }
    }

    componentDidMount() {
        this.setState({
            chartData: {
                labels: this.props.data.map(d => d.time),
                datasets: [{
                    label: this.props.label,
                    data: this.props.data.map(d => d.price),
                    fill: 'none',
                    backgroundColor: this.props.color,
                    pointRadius: 1,
                    borderColor: this.props.color,
                    borderWidth: 5,
                    lineTension: 0
                }]
            }
        });
    }

    render() {
        return (
            <Line ref={(reference) => this.chartReference = reference } data={this.state.chartData}/>
        );
    }
};

export default BitcoinChart;