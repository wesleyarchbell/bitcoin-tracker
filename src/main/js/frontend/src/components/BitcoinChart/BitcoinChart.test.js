import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import BitcoinChart from './BitcoinChart';

describe("BitcoinChart", () => {

    const dataMap = [
        {
            price: 3435.22,
            time: "2012-01-1 01:01:01"
        }
    ];

    it("Render chart", () => {
        const chart = shallow(<BitcoinChart data={dataMap} label="Test Chart"/>);            
        expect(chart).toMatchSnapshot();
    });
});