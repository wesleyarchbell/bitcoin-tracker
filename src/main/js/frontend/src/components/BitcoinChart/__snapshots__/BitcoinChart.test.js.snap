// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BitcoinChart Render chart 1`] = `
<Line
  data={
    Object {
      "datasets": Array [
        Object {
          "backgroundColor": undefined,
          "borderColor": undefined,
          "borderWidth": 5,
          "data": Array [
            3435.22,
          ],
          "fill": "none",
          "label": "Test Chart",
          "lineTension": 0,
          "pointRadius": 1,
        },
      ],
      "labels": Array [
        "2012-01-1 01:01:01",
      ],
    }
  }
/>
`;
